source/effect_codegen_glsl.cpp

```diff
@@ -758,42 +749,29 @@ class codegen_glsl final : public codegen
 	{
 		const id res = make_id();
 
-		std::string &code = _blocks.at(_current_block);
-
 		// Struct initialization is not supported right now
 		if (type.is_struct())
 		{
+			std::string &code = _blocks.at(_current_block);
+
 			code += '\t';
 			write_type(code, type);
 			code += ' ' + id_to_name(res) + ";\n";
 			return res;
 		}
 
-		code += "\tconst ";
-		write_type(code, type);
-		code += ' ' + id_to_name(res);
-
-		if (type.is_array())
-			code += '[' + std::to_string(type.array_length) + ']';
-
-		code += " = ";
+		std::string code;
 		write_constant(code, type, data);
-		code += ";\n";
+		define_name<naming::formula>(res, code);
 
 		return res;
 	}
 
-	id   emit_unary_op(const location &loc, tokenid op, const type &res_type, id val) override
+	id   emit_unary_op(const location &, tokenid op, const type &res_type, id val) override
```

Comment:
I'm a bit on the fence on whether it is worth throwing out line information for better readability ... But I can see the appeal and it might even improve HLSL compile times, so will probably go for it.